src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ([a, b, c].some((side) => side <= 0)) {
+      throw new Error('ERROR: invalid sides length (some length is <= than 0)');
+    }
+
+    const lengthSideСheck: boolean = a >= b + c || b >= a + c || c >= a + b;
 
+    if (lengthSideСheck) {
+      throw new Error(
+        'ERROR: invalid sides length '
+        + '(the longest side of a triangle is >= than a sum of two others)',
+      );
+    }
+  }
+
+  getArea():number {
+    const semiPerimeter: number = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiPerimeter
+        * (semiPerimeter - this.a)
+        * (semiPerimeter - this.b)
+        * (semiPerimeter - this.c),
+    );
+
+    return Math.round(area * 100) / 100;
```

Comment:
Optional: you round with the same logic twice over the code. You also could create a separate helper function - `rounder` for example for this purpose.

Reply:
👍